src/hotspot/cpu/aarch64/gc/shared/barrierSetAssembler_aarch64.cpp

```diff
@@ -288,8 +288,7 @@ void BarrierSetAssembler::c2i_entry_barrier(MacroAssembler* masm) {
   __ stp(r10, r11, Address(__ pre(sp, -2 * wordSize)));
```

Comment:
No need to save r11 then?

Reply:
"As resolve_weak_handle may call into the VM r11 may still be clobbered. "
-- I don't quite understand this.  If this is the case, shouldn't r11 be saved & restored immediate before and after the VM call together with other live caller-save registers?

Reply:
Maybe? Actually looking at it a second time it seems like r0-r7 and r9-r17 are pushed to the stack at the vm call I was looking at. 
I am probably missing something. I do not fully understand the reason for pushing the registers at this point, not earlier or later. Would have to investigate further. I do however think that such a change belongs to a separate issue/PR. This PR should not change the instruction emission, only which registers are used.

Reply:
I think the reason is that register r11 was used as temp register (and thus gets clobbered) by resolve_weak_handle previously. That's why this register was saved/restored here. But this has changed since it not used here anymore with your change. I am OK if you want to handle that in another PR. Just reminds. Thanks.

Reply:
I looked into it some further. You are correct. The reason I thought it was important was because this change hard crashed.
```diff
+ __ str(r10, Address(__ pre(sp, -wordSize)));
- __ stp(r10, r11, Address(__ pre(sp, -2 * wordSize)));
```
But I guess it is because I am misaligning the stack or there is some other reason `sp` cannot be bumped by `8`, only `16`. As this change is fine.
```diff
+ __ str(r10, Address(__ pre(sp, -2 * wordSize)));
- __ stp(r10, r11, Address(__ pre(sp, -2 * wordSize)));
```
Maybe using the RegSets and `push` would do this correctly, or it is some special invariant for these adapters.

Reply:
Fix using `MacroAssembler::push(RegSet, sp);`